CMakeLists.txt

```diff
@@ -85,6 +85,26 @@ endif(WITH_CCACHE)
 
 option(WITH_SCCACHE "Build with sccache.")
 if(WITH_SCCACHE)
+  if(NOT FORCE_MAX_JOBS)
+    execute_process(
+      COMMAND "sccache" "--dist-status"
+      OUTPUT_VARIABLE sccache_dist_status
+      RESULT_VARIABLE sccache_dist_status_result
+      COMMAND_ERROR_IS_FATAL ANY
+    )
+    message(STATUS "sccache --dist-status: ${sccache_dist_status}")
+    string(
+      JSON sccache_cores
```

Comment:
`string(JSON ...)` was introduced in CMake 3.19, see https://cmake.org/cmake/help/latest/command/string.html#json . i guess there're at least two approach to tackle this.

1. check for the cmake version before using it, and use a relatively safe `sccache_cores`, and print a warning to explain that we are using a wild guess.
2. bump up the minimal required cmake version to 3.19 in a dedicated commit. but please check if all distros we use in our CI provide this version.

Reply:
Thanks for catching this! I am quite new to CMake and ought to have thought to check on that feature's availability. It looks like ubuntu 20.04 is the only CI distro that would be incompatible. I've tweaked the logic so that if cmake is too old, we just don't try to make a max jobs estimate.